torch/distributed/pipelining/stage.py

```diff
@@ -102,7 +102,7 @@ def __init__(
             group (Optional[dist.ProcessGroup]): The process group to use for communication.
                 If `None`, the default process group will be used.
                 Default: `None`.
-            dw_builder (Optional[Callable[[], Callable[[], None]]): If provided, dw_runner is a builder function
+            dw_builder (Optional[Callable[[], Callable[..., None]]): If provided, dw_runner is a builder function
                 that will build a new dw_runner function that will run parts of module backward that were intentionally
```

Comment:
... otherwise, a dw_runner will be generated automatically by traversing the autograd graph.  (See Note: Automatic dW/dInput partitioning)

You really should add such a note that explains the algorithm e2e